### DISCUSSION TITLE
How do I deploy Node.js on the server and run it?

### DISCUSSION THREAD

**Discussion by RuoyuLiu-f** [community/community#132340] [npm] [answered]
[This project](https://github.com/burakorkmez/mern-chat-app)

**Comment by naiyannoor** [comment, marked as answer]
Certainly! Deploying a Node.js application on a server involves several steps, but fear not—I’ll guide you through the process. Let’s get your Node app up and running! 🚀

Choose a Server Provider: First, you’ll need a server to host your Node.js application. DigitalOcean is a popular choice for its simplicity and cost-effectiveness. You can create a “Droplet” (a virtual machine) there. If you haven’t already, sign up for a DigitalOcean account.
Create a DigitalOcean Project:
Log in to your DigitalOcean account.
Click on the “New Project” button in the left-side menu.
Give your project a name (anything you like).
You can skip moving resources for now; we’ll create the Droplet later.
Create a Droplet:
Click the “Get Started” button.
Choose an Ubuntu image (e.g., Ubuntu 20.04 LTS) as your Droplet’s base.
You can also select a Dedicated CPU if needed.
Follow the prompts to create your Droplet.
Access Your Server:
Once your Droplet is ready, you’ll receive an IP address.
Use SSH to connect to your server. For example:
ssh root@your_droplet_ip

Install Dependencies:
Update your server:
sudo apt update
sudo apt upgrade

Install Node.js and npm:
sudo apt install nodejs npm

Install PM2 (Process Manager):
PM2 will keep your Node app running even after you log out:
sudo npm install -g pm2

Upload Your Node.js App:
You can use SCP or SFTP to transfer your app files to the server.
Place your app in a directory (e.g., /var/www/myapp).
Run Your Node.js App with PM2:
Navigate to your app directory:
cd /var/www/myapp

Start your app using PM2:
pm2 start app.js

To ensure it starts automatically on server reboot:
pm2 save
pm2 startup

Set Up Nginx as a Reverse Proxy:
Install Nginx:
sudo apt install nginx

Create an Nginx configuration file for your app (e.g., /etc/nginx/sites-available/myapp).
Enable the site:
sudo ln -s /etc/nginx/sites-available/myapp /etc/nginx/sites-enabled/

Test Nginx configuration:
sudo nginx -t

Reload Nginx:
sudo systemctl reload nginx

Secure Your App with SSL (Optional):
Use Let’s Encrypt to obtain an SSL certificate for your domain.
And there you have it! Your Node.js app should now be accessible via your server’s IP address or domain name. 🎉 Remember to adapt these steps to your specific project and server environment. If you encounter any issues, feel free to ask—I’m here to help! 😊

Sources:

[Deploy a Node.js App – From Server Setup to Production](https://www.freecodecamp.org/news/deploy-nodejs-app-server-to-production/)[1](https://www.freecodecamp.org/news/deploy-nodejs-app-server-to-production/)
[How to deploy a Node.js application to a server with Nginx](https://maksymkotov.com/blog/how-to-deploy-an-node-js-application-to-a-server-with-nginx/)[2](https://maksymkotov.com/blog/how-to-deploy-an-node-js-application-to-a-server-with-nginx/)
[Best practice for Node.js deployment](https://stackoverflow.com/questions/50693283/best-practice-for-nodejs-deployment-directly-moving-node-modules-to-server-or)[3](https://stackoverflow.com/questions/50693283/best-practice-for-nodejs-deployment-directly-moving-node-modules-to-server-or)
[Set Up a Node.js Application for Production on Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-set-up-a-node-js-application-for-production-on-ubuntu-20-04)[4](https://www.digitalocean.com/community/tutorials/how-to-set-up-a-node-js-application-for-production-on-ubuntu-20-04)
Feel free to ask if you need further assistance or clarification! 😊

**Reply by RuoyuLiu-f** [reply]
Thank you.It's work now.
